Introduce Hindi Translation for Chapter-1

shaavan opened the pull request:
resolves #337

Adds translation for Chapter-1

shaavan commented:
Looking good so far!

**Screenshot:**

<img width="1491" alt="image" src="https://user-images.githubusercontent.com/85434418/229349279-8a7cf32b-6e2c-4f29-a88c-1c42cadd9fea.png">

shaavan commented:
It's Ready!!

<img width="1491" alt="image" src="https://user-images.githubusercontent.com/85434418/229534550-524b1d98-3add-440c-aca7-9aedd03d2a77.png">

shaavan commented:
@vinayaksh42

Please take a look through the introduced translations to cross-check the introduced translations!

vinayaksh42 reviewed (changes requested):
The translation for Hindi LGTM! Especially the homepage looks amazing!
Just a few things can be improved that I noticed on the /chapters page:
<img width="1073" alt="Screenshot 2023-04-03 at 9 24 32 PM" src="https://user-images.githubusercontent.com/54861487/229563666-efff777e-6fc1-4896-badc-f0f01f3e472c.png">
The link for the feedback seems to be breaking up, you can check out the English translation file (`en.ts`). Another thing that could be improved is converting the Heading `Chapters` and `Explore the mysteries of Satoshi and learn about Bitcoin along the way.` to Hindi.

vinayaksh42 commented on i18n/locales/hi.ts:
```diff
@@ -0,0 +1,320 @@
+  demo_disclaimer: {
+    title: 'यह एक डेमो है',
+    description:
+      'हम अभी भी इस परियोजना के प्रारंभिक चरण में हैं। अध्याय 1 कार्यात्मक है, शेष विकास के अधीन है। इसे आज़माएं और हमें बताएं कि आप क्या सोचते हैं, हमें <Link href="https://docs.google.com/forms/d/e/1FAIpQLSf1xpNqUYJyvYL5IZDnxy78273pkqzfYW2Hf91H4Do4KHgy9g/viewform?usp=sf_link" className="underline">आपकी प्रतिक्रिया</लिंक> अच्छी लगेगी।"',
```
you accidentally wrote `</Link>` as `</लिंक>`

shaavan commented:
Addressed @vinayaksh42 comment

- Fixed the <Link> tag on the `/chapters` page.

benalleng commented:
I made a change in `/components/LangSwitch.tsx` in #379 so that we don't have to keep adding new languages to make it easier for future contributors

shaavan commented:
Updated from 23b38204b7335541900d0ac3b9e71bdb42c7640d to cfbfd2d1eac61b53da6997d57f3c63493f63aeff ([pr345.02](https://github.com/shaavan/saving-satoshi/commits/pr345.02) -> [pr345.03](https://github.com/shaavan/saving-satoshi/commits/pr345.03))
Addressed @benalleng comment

**Changes:**
1. Rebased over master and resolved merge conflicts.
2. Introduced translation for chapter-2.

vinayaksh42 reviewed (changes requested):
the PR seems to be having build issues, @shaavan can you check what might be causing it?

shaavan commented:
I can compile my branch successfully on my local system using both `npm run build` and `npm run start`.

<details>
<summary>
<strong>Screenshot:</strong>
</summary>
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/85434418/232199271-ccdcd82f-4803-4ff9-b785-380d6ddb74ac.png">
</details>

@afterburn

Could you please check the possible issue with the Vercel built for this branch? Thanks!

benalleng commented:
Build succeeded on my end too

vinayaksh42 commented:
not sure what might be causing the issue because the build seems to be running fine on my system as well. One thing that I noticed when I ran `npm run build` on this branch and the master branch is the size of `First load JS`, it seems to be double compared to the master branch.

** ScreenShots **
Master branch:
<img width="850" alt="Screenshot 2023-04-18 at 1 14 04 PM" src="https://user-images.githubusercontent.com/54861487/232707731-eded6536-a92e-4f68-a6d2-0447c3af8551.png">

This branch:
<img width="829" alt="Screenshot 2023-04-18 at 1 13 01 PM" src="https://user-images.githubusercontent.com/54861487/232707878-0eb395aa-5040-4bbf-a012-bbbf2bdc58fb.png">

benalleng commented:
@shaavan I made a test build in #384 that succeeded, you might want to merge with master and see if there was some old state causing this build to fail

shaavan commented:
Updated from cfbfd2d1eac61b53da6997d57f3c63493f63aeff to aef3850163dd5b790146392b2147925eb6fe87c0 ([pr345.03](https://github.com/shaavan/saving-satoshi/commits/pr345.03) -> [pr345.04](https://github.com/shaavan/saving-satoshi/commits/pr345.04))
Addressed @benalleng suggestion

Changes:
1. Rebased over master.

ecurrencyhodler commented:
Success!
